src/main/java/uk/gov/hmcts/cft/idam/testingsupportapi/service/NotificationsService.java

```diff
@@ -0,0 +1,119 @@
+package uk.gov.hmcts.cft.idam.testingsupportapi.service;
+
+import com.google.common.annotations.VisibleForTesting;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.cft.idam.api.v2.common.error.SpringWebClientHelper;
+import uk.gov.service.notify.Notification;
+import uk.gov.service.notify.NotificationClient;
+import uk.gov.service.notify.NotificationClientException;
+import uk.gov.service.notify.NotificationList;
+
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@Slf4j
+@Service
+public class NotificationsService {
+
+    private static final Pattern NOTIFY_KEY_REFERENCE_REGEX = Pattern.compile("^[a-z]*[_a-z]*[_a-z]*");
+    private static final Pattern OLDER_THAN_REGEX = Pattern.compile("older_than=([A-Za-z0-9\\-]+)&?");
+
+    private static final String ALL_STATUSES = null;
+    private static final String EMAIL_TYPE = "email";
+    private static final int MAX_PAGES = 20;
```

Comment:
What's the rationale for setting this to 20?

Reply:
Big but not too big. Realistically we probably won't be searching for anything more than a few minutes old. Will probably paramaterise at some point, but just wanted to see how it goes like this for now.